app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationActivityV2.kt

```diff
@@ -298,6 +298,13 @@ class ConversationActivityV2 : PassphraseRequiredActionBarActivity(), InputBarDe
     private val reverseMessageList = false
 
     private val adapter by lazy {
+
+        // To prevent repeated attachment download jobs being spawned we'll keep a set of what
+        // attachmentId / mmsId pairs we've already attempted to download and only spawn the job
+        // if we haven't already done so. Without this then when the retry limit for a failed job
+        // hits another job is immediately spawned (endlessly).
+        var alreadyAttemptedAttachmentDownloadPairs = mutableSetOf<Pair<Long, Long>>()
```

Comment:
why Pairs? is one of these ids sufficient?

Reply:
I just thought it guarantees uniqueness, but have modified so that we only keep track of the attachmentIds and it seems to be fine with that.